File: app/src/main/java/com/zhiyong/tingxie/ui/word/WordViewModelFactory.java
package com.zhiyong.tingxie.ui.word;

import android.app.Application;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import androidx.annotation.NonNull;

public class WordViewModelFactory implements ViewModelProvider.Factory {
    private final Application mApplication;
    private final long mQuizId;

    public WordViewModelFactory(Application application, long quizId) {
        mApplication = application;
        mQuizId = quizId;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new WordViewModel(mApplication, mQuizId);
    }
}

File: app/src/main/java/com/zhiyong/tingxie/ui/word/WordViewModelFactory.kt
package com.zhiyong.tingxie.ui.word

import com.zhiyong.tingxie.Repository
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

@Suppress("UNCHECKED_CAST")
class WordViewModelFactory(
  private val repository: Repository,
  private val mQuizId: Long
) : ViewModelProvider.NewInstanceFactory() {
  override fun <T : ViewModel> create(modelClass: Class<T>): T {
    return WordViewModel(repository, mQuizId) as T
  }
}
